radicle-node/src/service/message.rs

```diff
@@ -173,46 +166,119 @@ impl wire::Decode for NodeAnnouncement {
     }
 }
 
+/// Node announcing project refs being created or updated.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct RefsAnnouncement {
     /// Repository identifier.
     pub id: Id,
     /// Updated refs.
     pub refs: Refs,
+    // TODO: Add timestamp
+}
+
+/// Node announcing its inventory to the network.
+/// This should be the whole inventory every time.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct InventoryAnnouncement {
+    pub inventory: Vec<Id>,
+    pub timestamp: Timestamp,
+}
+
+/// Announcement messages are messages that are relayed between peers.
+#[derive(Clone, PartialEq, Eq)]
+pub enum AnnouncementMessage {
+    /// Inventory announcement.
+    Inventory(InventoryAnnouncement),
+    /// Node announcement.
+    Node(NodeAnnouncement),
+    /// Refs announcement.
+    Refs(RefsAnnouncement),
 }
```

Comment:
What about naming them 'gossip' or 'relayed' messages?  Describing how they are treated as a message type instead of how they are currently used.